src/components/editor/metaform-editor-right-drawer.tsx

```diff
@@ -28,8 +29,59 @@ const MetaformEditorRightDrawer: React.FC<Props> = ({
   const [ tabIndex, setTabIndex ] = React.useState(0);
   const [ disabledValue, setDisabledValue ] = React.useState<boolean>(false);
   const [ requiredConditionField, setRequiredConditionField ] = React.useState<string>("");
-  const [ requiredConditionFieldBoolean, setRequiredConditionFieldBoolean ] = React.useState<string>("");
+  const [ requiredConditionEqualsValue, setRequiredConditionEqualsValue ] = React.useState<string>("");
   const [ requiredConditionInfoField, setRequiredConditionInfoField ] = React.useState<boolean>(true);
+  const [ sectionSelection, setSectionSelection] = React.useState<boolean | undefined>(false);
+  const [ selectedComponent, setSelectedComponent] = React.useState<string | undefined>("");
+  const [ metaFormFieldSection, setMetaFormFieldSection] = React.useState<MetaformField | MetaformSection>();
+  const [ metaformSectionTitle, setMetaFormSectionTitle] = React.useState<string | undefined>("");
+
+  /**
+   * Set values of Confidition field, switch and get selected component name
+   * 
+   */
+  const setVisiblityComponentValues = () => {
+    if (fieldIndex === undefined && sectionIndex !== undefined) {
+      setMetaFormFieldSection(pendingForm.sections![sectionIndex]);
+      setMetaFormSectionTitle(pendingForm.sections![sectionIndex].title);
+      const section = pendingForm.sections![sectionIndex].visibleIf;
+      if (section) {
+        setDisabledValue(true);
+        setRequiredConditionField(section.field!);
+        setRequiredConditionEqualsValue(section.equals!);
+      } else {
+        setDisabledValue(false);
+        setRequiredConditionField("");
+        setRequiredConditionEqualsValue("");
+      }
+    }
+    if (fieldIndex !== undefined && sectionIndex !== undefined) {
+      setMetaFormFieldSection(pendingForm.sections![sectionIndex].fields![fieldIndex]);
+      setSelectedComponent(pendingForm.sections![sectionIndex].fields![fieldIndex].name);
+      const field = pendingForm.sections![sectionIndex].fields![fieldIndex].visibleIf;
+      if (field) {
+        setDisabledValue(true);
+        setRequiredConditionField(field.field!);
+        setRequiredConditionEqualsValue(field.equals!);
+      } else {
+        setDisabledValue(false);
+        setRequiredConditionField("");
+        setRequiredConditionEqualsValue("");
+      }
```

Comment:
Use ternary operators to simplify this code and consider splitting two main if statements to two separate methods.